pkg/servicemesh/servicemesh-control-plane.go

```diff
@@ -0,0 +1,578 @@
+	glog.V(100).Infof("Initializing new serviceMeshControlPlane ControlPlaneBuilder structure with the following "+
+		"params: name: %s, namespace: %s", name, nsname)
+
+	builder := &ControlPlaneBuilder{
+		apiClient: apiClient,
+		Definition: &istioV2.ServiceMeshControlPlane{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the serviceMeshControlPlane is empty")
+
+		builder.errorMsg = "The serviceMeshControlPlane 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the serviceMeshControlPlane is empty")
+
+		builder.errorMsg = "The serviceMeshControlPlane 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithAllAddonsDisabled disable all addons to the serviceMeshControlPlane.
+func (builder *ControlPlaneBuilder) WithAllAddonsDisabled() *ControlPlaneBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	enablement := false
+
+	glog.V(100).Infof(
+		"Creating serviceMeshControlPlane %s with the all addons disabled", builder.Definition.Name)
+
+	if builder.Definition.Spec.Addons == nil {
+		builder.Definition.Spec.Addons = &istioV2.AddonsConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.Grafana == nil {
+		builder.Definition.Spec.Addons.Grafana = &istioV2.GrafanaAddonConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.Kiali == nil {
+		builder.Definition.Spec.Addons.Kiali = &istioV2.KialiAddonConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.Jaeger == nil {
+		builder.Definition.Spec.Addons.Jaeger = &istioV2.JaegerAddonConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.Prometheus == nil {
+		builder.Definition.Spec.Addons.Prometheus = &istioV2.PrometheusAddonConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.Stackdriver == nil {
+		builder.Definition.Spec.Addons.Stackdriver = &istioV2.StackdriverAddonConfig{}
+	}
+
+	if builder.Definition.Spec.Addons.ThreeScale == nil {
+		builder.Definition.Spec.Addons.ThreeScale = &istioV2.ThreeScaleAddonConfig{}
+	}
+
+	disableAllAddons := &istioV2.AddonsConfig{
+		Prometheus: &istioV2.PrometheusAddonConfig{
+			Enablement: istioV2.Enablement{
+				Enabled: &enablement,
+			},
+		},
+		Jaeger: &istioV2.JaegerAddonConfig{
+			Install: &istioV2.JaegerInstallConfig{
+				Ingress: &istioV2.JaegerIngressConfig{
+					Enablement: istioV2.Enablement{
+						Enabled: &enablement,
+					},
+				},
+			},
+		},
+		Grafana: &istioV2.GrafanaAddonConfig{
+			Enablement: istioV2.Enablement{
+				Enabled: &enablement,
+			},
+		},
+		Kiali: &istioV2.KialiAddonConfig{
+			Enablement: istioV2.Enablement{
+				Enabled: &enablement,
+			},
+		},
+	}
+
+	builder.Definition.Spec.Addons = disableAllAddons
+
+	return builder
+}
+
+// WithGrafanaAddon add grafana addons to the serviceMeshControlPlane.
+func (builder *ControlPlaneBuilder) WithGrafanaAddon(
+	enablement bool,
+	install *istioV2.GrafanaInstallConfig,
```

Comment:
Please add validation that install is not nil pointer.

Reply:
it could be empty, we don't have to configure each and every available for config option